pkg/ytdl/ytdl.go

```diff
@@ -55,11 +55,8 @@ func (dl YoutubeDl) Download(ctx context.Context, feedConfig *config.Feed, episo
 		return nil, errors.Wrap(err, "failed to get temp dir for download")
 	}
 
-	ext := "mp4"
-	if feedConfig.Format == model.FormatAudio {
-		ext = "mp3"
-	}
-	filePath := filepath.Join(tmpDir, fmt.Sprintf("%s.%s", episode.ID, ext))
+	// filePath with YoutubeDl template format
+	filePath := filepath.Join(tmpDir, fmt.Sprintf("%s.%s", episode.ID, "%(ext)s"))
```

Comment:
Does this guaranty that we'll download either mp3 or mp4 (and not other extensions) ?

Reply:
Before the ytdl refactoring the output template was exactly the same than in this PR
The output format, and therefore extension, is defined in the args:

audio: 
args = append(args, "--extract-audio", "--audio-format", "mp3", "--format", format

video:
format := "bestvideo[ext=mp4]+bestaudio[ext=m4a]/best[ext=mp4]/best"
args = append(args, "--format", format)